### PR TITLE
Add buf schema registry and api module 

### PR DESCRIPTION
**Pull request by mattverse**
Closes: #1678 

## What is the purpose of the change

Set up code generation for pulsar and the api module

## Brief Changelog

- Set up code generation for pulsar and the api module

## Testing and Verifying

Ensure `make proto-all` does not break, generates code for pulsar

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (yes / no)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (yes / no)
  - How is the feature or change documented? (not applicable   /   specification (`x/<module>/spec/`)  /  [Osmosis docs repo](https://github.com/osmosis-labs/docs)   /   not documented)

### CODE REVIEW

File: scripts/protocgen2.sh [hunk 1]
```diff
@@ -5,6 +5,7 @@ set -eo pipefail
 protoc_install_gopulsar() {
   go install github.com/cosmos/cosmos-proto/cmd/protoc-gen-go-pulsar@latest
   go install google.golang.org/grpc/cmd/protoc-gen-go-grpc@latest
+  go install github.com/cosmos/cosmos-sdk/orm/cmd/protoc-gen-go-cosmos-orm@latest #2>/dev/null
```
Comment (alexanderbez) [inline scripts/protocgen2.sh:8]:
```suggestion
  go install github.com/cosmos/cosmos-sdk/orm/cmd/protoc-gen-go-cosmos-orm@latest
```

**Review by alexanderbez — APPROVED** [review]
🚀

**Comment by ValarDragon** [comment]
Hrmm, can we see if its possible to get the buf api registry to only get made for packages we want to use ORM? (e.g. make a dummy file trying out ORM & hardcode that file for pulsar generation?)

**Comment by mattverse** [comment]
> Hrmm, can we see if its possible to get the buf api registry to only get made for packages we want to use ORM? (e.g. make a dummy file trying out ORM & hardcode that file for pulsar generation?)

I'm not sure if I understood u correctly. Are you suggesting we leave the api folder for the packages and modules we use and replace other files under the api directory that we're not using to empty dummy files? What's the reason behind this?

**Comment by siulee2** [comment]
🚀 I want this udpate.

**Comment by ValarDragon** [comment]
For context, does something in this update impact what you're doing?

Seems to me like it shouldn't really be of benefit to anyone who depends on this repo

**Comment by ValarDragon** [comment]
(Integrating the API module is a lot of work, and not considered stable in the SDK yet)

**Pull request closed without merging**